sphinx/ext/autodoc/__init__.py

```diff
@@ -413,6 +413,10 @@ def format_signature(self, **kwargs: Any) -> str:
         else:
             return ''
 
+    def format_extra_signatures(self) -> List[str]:
```

Comment:
TBH, I hesitate to add a new method to Documenter class because it is too fat class... It is very complex and hard to maintain for me. This is why I used mock module to hack `add_directive_header()`. -0 for adding a new method.

Reply:
Ok, I'll go back to your original solution with mock...